api/src/utils/get-accountability-for-role.test.ts

```diff
@@ -43,45 +42,50 @@ describe('getAccountabilityForRole', async () => {
 		expect(result).toStrictEqual({
 			admin: true,
 			app: true,
-			permissions: [],
+			roles: [],
 			role: null,
 			user: null,
 		});
 	});
 
 	test('get role from database', async () => {
-		const db = mockDatabase();
-
-		db['first'].mockReturnValue({
-			admin_access: 'not true',
-			app_access: '1',
-		});
+		vi.mocked(fetchRolesTree).mockResolvedValue([]);
+		vi.mocked(fetchGlobalAccess).mockResolvedValue({ admin: false, app: true });
 
 		const result = await getAccountabilityForRole('123-456', {
 			accountability: null,
 			schema: {} as any,
-			database: db as any,
+			database: {} as any,
 		});
 
 		expect(result).toStrictEqual({
 			admin: false,
 			app: true,
-			permissions: [],
+			roles: [],
 			role: '123-456',
 			user: null,
 		});
+
+		expect(fetchRolesTree).toHaveBeenCalledWith('123-456', {});
+		expect(fetchGlobalAccess).toHaveBeenCalledWith({}, []);
 	});
 
-	test('database invalid role', async () => {
-		const db = mockDatabase();
-		db['first'].mockReturnValue(false);
+	test('database invalid role returns default accountability', async () => {
+		vi.mocked(fetchRolesTree).mockResolvedValue([]);
+		vi.mocked(fetchGlobalAccess).mockResolvedValue({ admin: false, app: false });
+
+		const result = await getAccountabilityForRole('456-789', {
+			accountability: null,
+			schema: {} as any,
+			database: {} as any,
+		});
 
-		expect(() =>
-			getAccountabilityForRole('456-789', {
-				accountability: null,
-				schema: {} as any,
-				database: db as any,
-			}),
-		).rejects.toThrow('Configured role "456-789" isn\'t a valid role ID or doesn\'t exist.');
+		expect(result).toStrictEqual({
+			admin: false,
+			app: false,
+			roles: [],
+			role: null,
```

Comment:
Should this return `null` for `role`, the invalid role `'456-789'` or throw an error as it did before ?